{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Context Likelihood of Relatedness:\n",
    "\n",
    "## CLR Introduction\n",
    "The CLR algorithm is an extension of the relevance networks approach for identifying transcriptional regulatory interactions. The original relevance networks method used mutual information for scoring the similarity between the expression levels of two genes in a set of microarrays. A gene and a transcription factor are predicted to interact if the mutual information between the expression levels of the gene and itspotential regulator is above some set threshold.\n",
    "\n",
    "## Mutual Information between discrete variables\n",
    "Mutual Informaton gives the measurement of the reduction in uncertainty of parts of the outcome of a system given measurement of the other parts. \n",
    "\n",
    "Entropy given by the formula stated below is an ideal measure of uncertainty in the system.\n",
    "\n",
    "$ H(X) =  - \\sum_X P(X) log P(X) $\n",
    "\n",
    "Here, H(X) is the entropy of the random variable X, P(X) is the probability of it's occurence. \n",
    "\n",
    "For two variables X and Y, the joint entropy is given by the following formula:\n",
    "\n",
    "$ H(X,Y) =  - \\sum_X P(X,Y) log P(X,Y) $\n",
    "\n",
    "Here, H(X,Y) is the joint entropy of the random variables X and Y, P(X,Y) is the joint probability of their occurence.\n",
    "\n",
    "Mutual Information can be written in terms of the entropies and the joint entropy of 2 random variables.\n",
    "\n",
    "I(X,Y) = H(X) + H(Y) - H(X,Y) \n",
    "\n",
    "or\n",
    "\n",
    "I(X,Y) = H(X) - H(X|Y) \n",
    "\n",
    "or\n",
    "\n",
    "I(X,Y) = H(Y) - H(Y|X) \n",
    "\n",
    "## Mutual Information between continuous variables\n",
    "\n",
    "The above formulas are given for the calculation of MI between 2 discrete random variables. For continuous random variables there are 2 approaches. \n",
    "\n",
    "We can take the easy way out i.e binning the data into M discrete intervals $ a_i \\forall i \\in M$. For experimental data consisting of N measurements of a variable $ x_u \\forall u \\in N $, an indicator function $ \\Theta_i$ counts the number of data points within each bin. The probabilities are then estimated based on the relative frequency of occurence \n",
    "\n",
    "$ p(a_i) = \\frac{1 }{N} \\sum_{u=1}^{N} \\Theta_i(x_u) $\n",
    "\n",
    "with \n",
    "\n",
    "$ \\Theta_i(x_u) = \\begin{cases} \n",
    "                    1 \\mspace{3mu} if \\mspace{3mu} x_u \\in a_i \\\\\n",
    "                    0 \\mspace{3mu} otherwise\n",
    "                   \\end{cases} $\n",
    "                   \n",
    "For two variables, the joint probabilities are calculated from a multivariate histogram. It has been suggested to adaptively choose the sizes of the bins so that each bin has an uniform distribution of points. \n",
    "\n",
    "However, in this approach, each data point is assigned to one and only one bin. For datapoints near the boundary of the bin, small fluctuations may shift these to neighbouring bins. The positions of the borders can therefore strongly affect the resulting mutual informations. \n",
    "\n",
    "To overcome the drawbacks, datasets are assigned to multiple bins simultaneously. Thus the indicator function $\\Theta_i$ is extended to be a set of polynomial B-spline functions. \n",
    "\n",
    "\n",
    "The first step is the calculation of the knot vectors $t_i$.\n",
    "\n",
    "If the number of bins that each variable is assigned to is M and the degree of the polynomial of the spline is k, then the knot vectors will range from 0 to M-1+k. The knot vector values are as follows:\n",
    "\n",
    "$t_i = \\begin{cases}\n",
    "         0  \\mspace{4mu} if \\mspace{4mu} i < k \\\\\n",
    "         i - k + 1 \\mspace{4mu} if \\mspace{4mu} k \\leq i \\leq M-1 \\\\ \n",
    "         M-1-k+2 \\mspace{4mu} if \\mspace{4mu} i > M-1\n",
    "         \\end{cases}$\n",
    "        \n",
    "Next it is required to define z which is the normalized form of the variable values and will be used to calculate the B function. \n",
    "\n",
    "\n",
    "$z = x - x_{min} \\frac{M_x - k + 1}{x_{max} - x_{min}} $\n",
    "\n",
    "Then the weights of the values for every bin is calculated using the B-spline function $B_{i,k}$ which is given by the following formula.\n",
    "\n",
    "$B_{i,1}(z) = \\begin{cases}\n",
    "                1 \\mspace{4mu} if \\mspace{4mu} t_i \\leq z \\leq t_{i+1} \\\\\n",
    "                0 \\mspace{4mu} otherwise\n",
    "               \\end{cases}$\n",
    "               \n",
    "and the following recursive formula\n",
    "               \n",
    "$ B_{i,k}(z) = B_{i,k-1} (z) \\frac {z-t_i}{t_{i+k-1} - t_i} + B_{i+1,k-1}(z) \\frac{t_{i+k} - z}{t_{i+k} - t_{i+1}}$     \n",
    "\n",
    "Once the weighting coefficients for each x value are calculated,  we determine the probability for each bin $a_i$ from this formula: \n",
    "\n",
    "$ p(a_i) = \\frac{1}{N} \\sum_{u=1}^{N} B_{i,k} (z_u) $\n",
    "\n",
    "The joint probability function is given below:\n",
    "\n",
    "$p(a_i,b_j) = \\frac{1}{N} \\sum_{u=1}^{N} B_{i,k}(zx_u) B_{j,k} (zy_u) $\n",
    "\n",
    "With the probability and joint probability of the variables defined, we can determine the entropy of each variable, joint entropy of two variables and finally the mutual information between them.\n",
    "\n",
    "## CLR filtering step\n",
    "\n",
    "After computing the mutual information between regulators and their potential target genes, CLR calculates the statistical likelihood of each mutual information value within its network context. The algorithm compares the mutual information between a transcription factor/gene pair to the‘‘background’’distribution of mutual information scores for all possible transcription factor/gene pairs that include either the transcription factor or its target. The most probable interactions are those whose mutual information scores stand significantly above the background distribution of mutual information scores. \n",
    "\n",
    "The background distribution is constructed from two sets of MI values: ${MI_i}$ , the set of all the mutual information values for gene i(in row or column i), and ${MI_j}$, the set of all the mutual information values for gene j(inrow or column j). Because of the sparsity of biological regulatory networks, most MI scores in each row of the mutual matrix represent random background MI (e.g., due to indirect network relationships). We approximate this background MI as a joint normal distribution with $MI_i$ and $MI_j$ as independent variables, which provides a reasonable approximation to the empirical distribution of mutual information. Thus, the final form of our likelihood estimate becomes\n",
    "\n",
    "$ f(Z_i,Z_j) = \\sqrt{Z_i^2 + Z_j^2} $\n",
    "\n",
    "where $Z_i$ and $Z_j$ are the z-scores of $MI_{ij}$ from the marginal distributions, and $f(Z_i,Z_j)$ is the joint likelihood measure."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Inferelator\n",
    "\n",
    "Here, we  assume  that  the  expression  level  of  a  gene,  or  the  mean expression  level  of  a  group  of  co-regulated  genes  $y$,  is  influenced by the level of N other factors in the system: $X = (x_1, x_2 \\ldots x_N)$.  In  principle,  an  influencing  factor  can  be  of  virtually any  type  (for  example,  an  external  environmental  factor,  a small molecule, an enzyme, or a post-translationally modified protein).  We  consider  factors  for  which  we  have  measured levels under a wide range of conditions; in this work we use TF transcript levels and the levels of external stimuli as predictors   and   gene   and   bicluster   trancript   levels   as   the response. Methods for selecting which of these factors are the most  likely  regulators,  among  all  possible  regulatory  influence factors, are described below.\n",
    "\n",
    "$ \\tau \\frac{dy}{dt} = -y + g(\\beta . Z) $\n",
    "\n",
    "Here, $Z = (z_1[X], z_2[X]  \\ldots z_P[X])$ is a set of functions of the regulatory  factors  X.  The  coefficients  $\\beta_j$,  for  ${j =  1,2,\\ldots,P}$,describe the influence of each element of Z, with positive coefficients corresponding to inducers of transcription and nega-tive coefficients to transcriptional repressors.  The  choice $z_j(X) = x_j \\mspace{4mu} for \\mspace{4mu} (j = 1, 2 \\ldots P = N)$ amounts to the simple weighted linear combination of influencing factors $\\beta . Z = \\sum \\beta_j x_j $ . Various  functional  forms  can  be  adopted  for  the  function  g, called  the  'nonlinearity'  or  'activation'  function  for  artificial neural  networks,  and  the  'link'  function  in  statistical  modeling.  The  function  g often  takes  the  form  of  a  sigmoidal,  or logistic, activation function \n",
    "\n",
    "$ g(\\beta . Z) = \\frac{1}{1 + e^{-\\beta . z}}$\n",
    "\n",
    "Given our model formulation, for selecting subsets of predictors we adopt the L1  shrinkage  or  the  LASSO:\n",
    "\n",
    "$ (\\hat{\\alpha}, \\hat{\\beta} ) = argmin_{\\alpha,\\beta} {\\sum_{i=1}^{N} ( y_i - \\alpha - \\sum_{j=1}^{p} \\beta_j z_{ij} )^2 }$\n",
    "\n",
    "subject to: \n",
    "\n",
    "$ \\sum_{j=1}^p |\\beta_j| \\leq t |\\beta_{ols}| $\n",
    "\n",
    "where $\\beta_{ols}$ is the ordinary least square estimate of $\\beta$.  The shrinkage parameter t can range from 0 to 1. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# An Optimization Framework \n",
    "\n",
    "An optimization algorithm was used to construct the network.\n",
    "\n",
    "The network of regulatory influences for each organismwas generated by iteratively determining for each gene clusterthe maximum amount of experimental expression change thatcan be accounted for a given number of regulatory influences. $J$,$K$, and $T$ denote the total number of gene clusters, TF clusters,and time points, respectively. The experimental expression level of gene cluster $j$ at time point $t$ is represented as $X_{jt}$, and theexperimental expression level of TF cluster $k$ at time point $t$ is denoted as $TF_{kt}$. $C_{kj}$ is the interaction coefficient that describes the influence of TF cluster $k$ on gene cluster $j$.\n",
    "\n",
    "Gene expression was assumed to be a linear additive contribution of the set of influencing TFs. This can be postulated for each gene cluster $j$ and time point $t$ as shown in:\n",
    "\n",
    "$ \\frac{dX_{jt}}{dt} = \\sum_{k}^{K} C_{kj} TF_{kt}  \\mspace{3mu} \\forall j  \\mspace{3mu} \\in J \\mspace{3mu} \\& \\mspace{3mu}  \\forall \\mspace{3mu}  t \\mspace{3mu} \\in \\mspace{3mu} T $ \n",
    "\n",
    "Backward finite difference was used to approximate the rateof expression term (i.e., left-hand side) in the mixed integer linear program formulation shown above. Solving this formulation identifies the interaction coefficients ($Ckj$) that minimize the amount of change in gene cluster expression not explained by the regulatory influences. The formulation was solved iteratively for every gene cluster and a range of maximum regulatory influences.\n",
    "\n",
    "\n",
    "$\\begin{bmatrix}\n",
    "min \\sum_j^J \\sum_t^{T-1} S_{jt}^{+} + S_{jt}^{-} \\\\\n",
    "\\\\\n",
    "s.t.\\\\\n",
    "\\\\\n",
    "X_{jt+1} - X{jt} - \\Delta t \\sum_k^K C_{kj} TF_{kt+1} \\\\\n",
    "  = S_{jt}^{+} - S_{jt}^{-} \\mspace{3mu} \\forall t \\mspace{3mu} \\in \\mspace{3mu} {1, \\ldots, T - 1}\\\\\n",
    "\\\\\n",
    "-My_{kj} \\leq C_{kj} \\leq My_{kj} \\mspace{3mu} \\forall k \\in K \\\\\n",
    "\\\\\n",
    "\\sum_k^K y_{kj} \\leq Max \\mspace{3mu} Regulatory \\mspace{3mu} Influences\n",
    "\\end{bmatrix} \\forall \\mspace{3mu} j \\mspace{3mu} \\in \\mspace{3mu} J $\n",
    "\n",
    "The slack variables $S_{jt}^+$ and $S_{jt}^−$ represent the positive and negative deviations from experimental measurements. The binary variables, $y_{kj}$, are used to restrict the number of regulatory influences per gene cluster. If a transcription factor in TF cluster $k$ is in gene cluster $j$ then a regulatory interaction will be imposed for that TF cluster gene cluster pair by fixing $y_{kj}$ to one for that pair. In order to determine the number of regulatory influences per cluster, the percentage of experimental expression change accounted for by the network was calculated for the range from one to 11 regulatory influences for both organisms. For a given number of regulatory influences this percentage of explained expression variance is defined as the difference between the error (i.e., the sum of the slack variables) with zero regulatory influences and the given number of regulatory influences,divided by the error with zero influences. The number of regulatory influences was chosen for each cluster as the minimum number of influences that brought the percentage of explained expression change above 50%."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
